Sources/StreamChat/Database/DatabaseContainer_Tests.swift

```diff
@@ -175,6 +175,26 @@ class DatabaseContainer_Tests: StressTestCase {
         XCTAssertNil(testObject)
     }
     
+    func test_databaseContainer_createsNewDatabase_whenPersistentStoreFailsToLoad() throws {
+        // Create a new on-disc database with the test data model
+        let dbURL = URL.newTemporaryFileURL()
+        var database: DatabaseContainerMock? = try DatabaseContainerMock(
+            kind: .onDisk(databaseFileURL: dbURL),
+            modelName: "TestDataModel",
+            bundle: Bundle(for: DatabaseContainer_Tests.self)
+        )
+                
+        // Create a new database with the same url but totally different models
+        database = try DatabaseContainerMock(
+            kind: .onDisk(databaseFileURL: dbURL),
+            modelName: "TestDataModel2",
+            bundle: Bundle(for: DatabaseContainer_Tests.self)
+        )
+        
+        // Should re-create the database
+        XCTAssertNotNil(database)
```

Comment:
I think it's not possible for the program to actually reach this point when `database` is `nil`. So, this test basically tests that the DB init doesn't throw. So maybe we can even do something like this:
```swift
// Create a new on-disc database with the test data model
         let dbURL = URL.newTemporaryFileURL()
         _ = try DatabaseContainerMock(
             kind: .onDisk(databaseFileURL: dbURL),
             modelName: "TestDataModel",
             bundle: Bundle(for: DatabaseContainer_Tests.self)
         )

        // Create a new database with the same url but totally different models and assert it doesn't throw
        XCTAssertNoThrow(
            try DatabaseContainerMock(
                kind: .onDisk(databaseFileURL: dbURL),
                modelName: "TestDataModel2",
                bundle: Bundle(for: DatabaseContainer_Tests.self)
            )
        )
```

Reply:
Good catch! 🤦

Reply:
Done ✅